src/V3EmitCBase.h

```diff
@@ -75,11 +75,34 @@ class EmitCBaseVisitor VL_NOT_FINAL : public AstNVisitor {
         return modp == v3Global.rootp()->constPoolp()->modp();
     }
 
+    static bool isAnonOk(const AstVar* varp) {
+        return v3Global.opt.compLimitMembers() != 0  // Enabled
+               && !varp->isStatic()  // Not a static variable
+               && !varp->isSc()  // Aggregates can't be anon
+               && (varp->basicp() && !varp->basicp()->isOpaque());  // Aggregates can't be anon
+    }
+
+    enum class VarClass : uint8_t {
+        Io = 0,
+        Signal = 1,
+        Temporary = 2,
+        Parameter = 3,
+        Other = 4  // E.g.: interface ref
+    };
+
+    static VarClass varClass(const AstVar* varp) {
+        return varp->isIO()                                  ? VarClass::Io
+               : (varp->isSignal() || varp->isClassMember()) ? VarClass::Signal
+               : (varp->isTemp())                            ? VarClass::Temporary
+               : varp->isParam()                             ? VarClass::Parameter
+                                                             : VarClass::Other;
+    }
```

Comment:
Perhaps better to have the class enum as a Verilator-style class with enum inside, then this can be inside the class as can be an ascii() method to print the enum value as a string.

(Unfortunately C++ standard didn't allow methods in enum class).

Reply:
Without the prefixes there is not much use for this so dropped.